test/jdk/java/awt/Frame/FrameRepackTest.java

```diff
@@ -0,0 +1,177 @@
+ * @summary Test dynamically changing frame component visibility and repacking
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual FrameRepackTest
+ */
+
+public class FrameRepackTest  {
+
+    private static final String INSTRUCTIONS = """
+            There is a green frame with a menubar.
+            The menubar has one menu, labelled 'Flip'.
+            The menu has two items, labelled 'visible' and 'not visible'.
+            The frame also contains a red panel that contains two line labels,
+            ' This panel is always displayed' and ' it is a test'.
+
+            If you select the menu item 'Flip->visible', then another panel is
+            added below the red panel.
+            The added panel is blue and has yellow horizontal and vertical scrollbars.
+
+            If you select menu item 'Flip->not visible', the second panel
+            is removed and the frame appears as it did originally.
+
+            You can repeatedly add and remove the second panel in this way.
+            After such an addition or removal, the frame's location on the screen
+            should not have changed, while the size changes to accommodate
+            either just the red panel or both the red and the blue panels.
+
+            If you resize the frame larger, the red panel remains at the
+            top of the frame with its height fixed and its width adjusted
+            to the width of the frame.
+
+            Similarly, if it is present, the blue panel and its yellow scroolbars
+            remain at the bottom of the frame with fixed height and width adjusted
+            to the size of the frame.  But selecting 'visible' or 'not visible'
+            repacks the frame, thereby adjusting its size tightly to its panel(s).
+
+            Upon test completion, click Pass or Fail appropriately.
+            """;
+
+    public static void main(String[] args) throws Exception {
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .title("FrameRepackTest Instructions")
+                .instructions(INSTRUCTIONS)
+                .testTimeOut(5)
+                .rows(30)
+                .columns(45)
+                .build();
+
+        SwingUtilities.invokeAndWait(() -> {
+            FrameRepack frame = new FrameRepack();
+
+            PassFailJFrame.addTestWindow(frame);
+            PassFailJFrame.positionTestWindow(frame,
+                    PassFailJFrame.Position.HORIZONTAL);
+
+            frame.setVisible(true);
+        });
+
+        passFailJFrame.awaitAndCheck();
+    }
+
+}
+
+class FrameRepack extends Frame implements ActionListener {
+
+    Panel south;
+
+    public FrameRepack() {
+        super("FrameRepack");
+
+        // create the menubar
+        MenuBar menubar = new MenuBar();
+        this.setMenuBar(menubar);
+        // create the options
+        Menu flip = new Menu("Flip");
+        MenuItem mi;
+        flip.addSeparator();
```

Comment:
The separator as the first item in a drop-down menu looks weird and confusing, I propose removing it.